pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/StudyBuilder.java

```diff
@@ -337,14 +337,28 @@ private StudyDto getStudyOrInsert(Handle handle, long tenantId, long umbrellaId)
             if (passwordRedirectUrl != null && alreadyHasRedirectUrl) {
                 throw new DDPException("There is already a client with a password redirect URL. Currently only one is allowed.");
             }
+            String clientDomain = ConfigUtil.getStrIfPresent(clientCfg, "domain");
+            Long clientTenantId = null;
+            if (StringUtils.isNotBlank(clientDomain)) {
+                //lookup tenant
+                JdbiAuth0Tenant jdbiTenant = handle.attach(JdbiAuth0Tenant.class);
+                Auth0TenantDto dto = jdbiTenant.findByDomain(clientDomain);
+                if (dto == null) {
+                    throw new DDPException("Client Domain :" + clientDomain + " not found ");
+                }
+                clientTenantId = dto.getId();
+                LOG.info("Using client domain: {} ", clientDomain);
+            }
+
+            long finalTenantId = (clientTenantId != null) ? clientTenantId : tenantId;
             ClientDto clientDto = jdbiClient
-                    .findByAuth0ClientIdAndAuth0TenantId(clientId, tenantId)
+                    .findByAuth0ClientIdAndAuth0TenantId(clientId, finalTenantId)
```

Comment:
Is there a case where this could end up using the same `clientId` for two different tenants?

Reply:
I don't think so unless configured incorrectly